shub/utils.py

```diff
@@ -105,9 +105,16 @@ def patch_sys_executable():
             except NotFoundException:
                 # Explicitly ask for installing 2.7
                 raise NotFoundException("Please install Python 2.7")
+        # PyInstaller sets this environment variable in its bootloader. Remove
+        # it so the system-wide Python installation uses its own library path
+        # (this is particularly important if the system Python version differs
+        # from the Python version that the binary was compiled with)
+        orig_lib_path = os.environ.pop('LD_LIBRARY_PATH', None)
         sys.executable = py_exe
         yield
         sys.executable = orig_exe
+        if orig_lib_path:
+            os.environ[str('LD_LIBRARY_PATH')] = orig_lib_path
```

Comment:
str() output in py3 will be unicode string, but bytes in py2. :confused: 

and I just tried with this simple script and it works in py2 and py3

```
from __future__ import unicode_literals
import os
os.environ['foo'] = 'bar'
print(os.environ.get('foo'))
```